framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -471,42 +577,113 @@ private static Receiver parseParameter(String s, FlowExpressionContext context)
         return context.arguments.get(idx - 1);
     }
 
+    /**
+     * If the argument is in the list, return a corresponding formal paramter such as "#2".
+     * Otherwise, return the argument unchanged.
+     *
+     * @param s expression string
+     * @param argumentList list of arguments corresponding to formal parameters
+     * @return a formal parameter such as "#2", or the argument string
+     */
+    public static String asFormalParameter(String s, ArrayList<String> argumentList) {
+        if (argumentList.contains(s)) {
+            return "#" + Integer.toString(argumentList.indexOf(s) + 1);
+        } else {
+            return s;
+        }
+    }
+
     /**
      * Parse a method call. First of returned pair is a pair of method name and arguments. Second of
      * returned pair is a remaining string.
      *
      * @param s expression string
+     * @param argumentList list of arguments corresponding to formal parameters
```

Comment:
Of what method?
Also, arguments always correspond to formal parameters, so you can cut "corresponding to formal parameters".